data/campaigns/World_Conquest/lua/game_mechanics/wocopedia/help.lua

```diff
@@ -68,49 +49,71 @@ function wesnoth.wml_actions.wc2_show_wocopedia(cfg)
 		local str_des_era = _ "<b>Factions</b>\n The Word Conquest 2 era consists of faction that are build of pairs of mainline faction of which at one has a healer available (Drakes, Rebels and Loyalists), and one does not (Orcs, Dwarves and Undead) the recruilist is also organized in pairs so that you sometimes have to recruit a different units before you can recruit the units that you want. The available heroes, desertes and random leaders also depend on your factions, the items you can get do not depend on the faction you choose."
 		local str_cat_settings = _ "Settings"
 
+
+
+		local root_node = dialog:find("treeview_topics")
+		local details = dialog:find("details")
+
+		function gui.widget.add_help_page(parent_node, args)
```

Comment:
This probably shouldn't be shoved into the `gui.widget` namespace. Is there a really good reason for it? If not, I'd probably suggest making it a local function. On the other hand, even if there _is_ a really good reason, I wonder if we could come up with a better way?

Reply:
gui.widget is (more or less) the widget metatale , this way i can use it like widget:add_help_page.  I think its fine the way its done here tbh, its true that this function only works inside this file but i don't really see this as a problem.

Reply:
Ahh, I was wondering if that was the case. It's still not a great idea to be adding stuff to the widget metatable, but I guess it's okay since it's in a campaign rather than a multiplayer addon? Unless of course it's an MP campaign?

Maybe just assign `gui.widget.add_help_page = nil` in the postshow? At least then it won't leak out into any other addon, which wouldn't matter if this is single-player but does set a better example for anyone who _does_ want to do something similar in multiplayer. Calling it outside the scope of this dialog would never work anyway, right?

Reply:
> rather than a multiplayer addon? Unless

Its unrelated to single or multiplayer in particular since addons work in single and multi players.

> gui.widget.add_help_page = nil in the postshow?


well i'm not against doing this but i also don't see this as a problem.